### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by Tyxiel** [celzero/rethink-app#1732] [Q&A] [answered]
1. When i have advanced wireguard ON, can the custom DNS PROVIDER, see my IP?

2. Does the RETHINK APP download/update blocklist via proxy?

3. Why when ADVANCED DNS is ON, some apps fail to work?

**Comment by ignoramous** [comment, marked as answer]
1. Yes. This will change in `v055o`, the next version: #1510 
2. No. While Rethink itself can be routed through Rethink (as if it were any other app), when *Configure* -> *Network* -> *Loopback* is turned ON. Though, no firewall / domain / proxy rules are yet applied to it. We might someday support this.
3. Multiple issues. Hopefully, we've fixed most of them in `v055o`, and can graduate it from its "Experimental" status.

**Comment by Tyxiel** [comment]
Thanks for you help, this app is getting really good
